### PR TITLE
docs(checkbox): add checkbox storybook docs

### LINKED ISSUE #6628: Update docs for Checkbox
This issue is to update the stories that we have for the <Component> component in our storybook. This work includes the following:

- [ ] Update each story in `<Component>-story.js` to Component Story Format
  - There is an optional codemod that you can use from storybook:
  - `npx -p @storybook/cli@next sb migrate storiesof-to-csf --glob='packages/react/src/<Component>/<Component>-story.js'`
- [ ] Update stories in the existing file, including:
  - [ ] Create a default example named after the component without any knobs (should match the first example in the overview section)
  - [ ] Clean-up or streamline any stories that seem to need it
  - [ ] Create a `Playground` story that uses knobs for the component
    - For any other components that use knobs, add a `<Component> Playground` story
- [ ] Create an `<Component>.mdx` file and fill it out according to the template below
  - You should import the `mdx` file in your story and pass it as an option to the story config (see example below)
  - You can refer to stories by their story id from `<Component>-story.js` in `<Component>.mdx`
- [ ] Document any usage mishaps for v11

<details>
<summary>Template</summary>

```mdx
import { Story, Preview, Props } from '@storybook/addon-docs/blocks';

# <Component name>

## Overview
<!--
This should include the default use-case(s) for a component, at most 2.
This should be the most common way the component is used.
It should summarize what this common is for and provide snippets for how
to accomplish these common use-cases
-->

<Preview>
  <Story id="story-id" />
</Preview>

## [Use case, for example, Skeleton state or Sorting]
<!--
Each use-case of this component becomes its own section. For example,
if the component had a skeleton then you would name this section Skeleton.

If the component had a specific piece of functionality or behavior, like
how you can sort with a Data Table, then Sorting would be the name of this
section.

This section should include an explainer for the use-case and when to use it,
along with a code snippet that someone could bring into their project.
-->

## Component API

<Props />

### [Component name] [prop name]
<!--
Used to clarify prop usage, provide additional info or snippets
-->


## References
<!--
A place for references, links, or a way clarify points made earlier
in the documentation.
-->

## Feedback

Help us improve this component by providing feedback through, asking questions on Slack, or
updating this file on GitHub
[GitHub](https://github.com/carbon-design-system/carbon/edit/master/packages/react/src/components/<Component>/<Component>.mdx).
```
</details>

<details>
<summary>Configure story to use MDX page</summary>

```bash
import mdx from './Component.mdx';

export default {
  title: 'Component',
  component: Component,
  parameters: {
    docs: {
      page: mdx,
    },
  },
}
```
</details>

### PR DESCRIPTION
**Pull request by aledavila**
Closes #6628

Add mdx file to checkbox and update storybook examples.

#### Changelog

**New**

- mdx

**Changed**

- storybook variants

#### Testing / Reviewing

Make sure examples are correct.

### CODE REVIEW

File: packages/react/src/components/Checkbox/Checkbox.mdx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# Checkbox
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Skeleton state](#skeleton-state)
+- [Component API](#component-api)
+- [Feedback](#feedback)
+
+## Overview
```
Comment (joshblack) [inline packages/react/src/components/Checkbox/Checkbox.mdx:13]: Should we include an example where we show how to do a "controlled" example? e.g.

```js
function ExampleComponent() {
  const [isChecked, setIsChecked] = useState(false);
  return <Checkbox checked={isChecked} onChange={setIsChecked} />
}
```

File: packages/react/src/components/Checkbox/Checkbox.mdx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+You can build a checkbox using the `Checkbox` in combination with a `<fieldset>` element to group controls and `<legend>` element for the label.
+
+<Preview>
+  <Story id="checkbox--checkbox" />
+</Preview>
+
+## Skeleton state
+
+You can use the `CheckboxSkeleton` component to render a skeleton variant of a checkbox/ This is useful to display while content in your breadcrumb is being fetched from an external resource like an API.
```
Comment (tw15egan) [inline packages/react/src/components/Checkbox/Checkbox.mdx]: Referencing `Breadcrumb` here. Doesn't seem applicable to the Checkbox, the skeleton states would probably be used on initial page load

File: packages/react/src/components/Checkbox/Checkbox.mdx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+<Props />
+
+### Checkbox defaultChecked
+
+You can use the `defaultChecked` prop to specify weather a checkbox is checked by default or not on page load.  
+
+```jsx
+<fieldset className={`${prefix}--fieldset`}>
+  <legend className={`${prefix}--label`}>Checkbox heading</legend>
```
Comment (tw15egan) [inline packages/react/src/components/Checkbox/Checkbox.mdx]: Can we remove the `fieldset` and `legend` elements from these examples? Seem to be adding a lot of clutter to the code samples and taking away from the prop being discussed.

**Review by tw15egan — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Checkbox/Checkbox.mdx (hunk 1 above)
Comment (aledavila) [reply packages/react/src/components/Checkbox/Checkbox.mdx:13]: Thats a good idea. Simple and clean example. I can add this.

**Comment by aledavila** [comment]
@joshblack @tw15egan Ready for review again

**Review by dakahn — APPROVED** [review]
Seems great! 🏄

File: packages/react/src/components/Checkbox/Checkbox.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+You can use the `defaultChecked` prop to specify weather a checkbox is checked by default or not on page load.  
+
+```jsx
+<Checkbox defaultChecked labelText={`Checkbox label`} id="checkbox-label-1" />
+<Checkbox defaultChecked labelText={`Checkbox label`} id="checkbox-label-2" />
+```
+
+### Checkbox id
+
+The `labelText` should be used to provide each checkbox input with a unique id and is a required prop.
```
Comment (tw15egan) [inline packages/react/src/components/Checkbox/Checkbox.mdx]:
```suggestion
The `id` should be used to provide each checkbox input with a unique id and is a required prop.
```

**Comment by aledavila** [comment]
@tw15egan done

File: packages/react/src/components/Checkbox/Checkbox.mdx (hunk 1 above)
Comment (tw15egan) [reply packages/react/src/components/Checkbox/Checkbox.mdx:13]: @aledavila was this ever added? See https://github.com/carbon-design-system/carbon/pull/6710/files#diff-3288281412b9946109165a563744f1c0R199-R214

Comment (aledavila) [reply packages/react/src/components/Checkbox/Checkbox.mdx:13]: @tw15egan yes it’s under overview in controlled example. Do you think this should instead be under API?

Comment (tw15egan) [reply packages/react/src/components/Checkbox/Checkbox.mdx:13]: Ah whoops, my mistake! missed that 👍 

Looks good where it is, I just wanted to make sure we can link directly to it

**Review by tw15egan — APPROVED** [review]

**Pull request merged**